### ISSUE TITLE
Address derivation change

### ISSUE THREAD

**Issue by punwai** [MystenLabs/sui#3521]
Signatures will now change from [SIGNATURE_BYTES, PUBLIC_KEY_BYTES] to [FLAG (1 byte), SIGNATURE_BYTES, PUBLIC_KEY_BYTES].

The FLAG for edDSA will currently be 0x00, and will currently be the only supported signature scheme.

The derivation for addresses have now changed. Addresses are still 20 bytes long. However, the address will be calculated as:

Hash(FLAG (1 byte) || PublicKey)[:20]

**Comment by Clay-Mysten** [comment]
Adding the usual doc leads for help determining the scope of this change. Also, I saw reference to config files needing updating. Please include those and other related pull requests here so we can see what's changing there. Thanks!

**Comment by Clay-Mysten** [comment]
Starting to sleuth in the face of little background:
```
claymurphy@Clays-MacBook-Pro sui % grep -i -r ed25519 doc/
doc//src/learn/architecture/consensus.md:Narwhal is implemented using [Tokio](https://github.com/tokio-rs/tokio), [RocksDB](https://github.com/facebook/rocksdb/) and generic cryptography. The cryptography contains implementations of node signing using BLS12-377, BLS12-381, and Ed25519.
claymurphy@Clays-MacBook-Pro sui % grep -i -r EdDSA doc/  
doc//src/learn/sui-security.md:signature key (currently using the EdDSA algorithm). This signature key can be kept private by the user and not be shared with 
claymurphy@Clays-MacBook-Pro sui % 
```

Not many references in the existing docs. Now starting to look at the seemingly related and long in conversation:
https://github.com/MystenLabs/sui/pull/3191

@punwai , I am suggesting it is time you become familiar with docs.sui.io and help determine where this change fits in. Thanks!

**Comment by punwai** [comment]
@Clay-Mysten Sorry for the delay. Just letting you know that so far, everything has been changed in one PR (sdk, core etc.), but I'll keep posting here as fixes come along the way.

**Comment by punwai** [comment]
https://github.com/MystenLabs/sui/pull/3595

**Comment by punwai** [comment]
@Clay-Mysten I checked the website, and it doesn't look like this really affects any information there at the moment. I think we'll eventually need to do a push to allow wallets to support edDSA and ecDSA, but still waiting for a green light on that.

**Comment by Clay-Mysten** [comment]
Adding @punwai to this issue as an assignee. Pun, I will need some more guidance than provided to continue. Please take a look at docs.sui.io and advise. Thanks!

**Comment by Clay-Mysten** [comment]
> #3595

From this, we should evaluate for updates:
https://docs.sui.io/build/json-rpc
https://docs.sui.io/build/sui-json

@patrickkuo may have other input on where docs are affected.

**Comment by punwai** [comment]
Hi Clay, to be clear about the scope of the changes - Addresses are now derived differently, but are in the exact same format. I looked through the docs and I was not able to find a place where the derivation of the address is explained, and hence, I don't think there is anything to update. If there is a section on how the address is derived from the public key, we should update it to say that it is now Hash(Flag || PublicKey), with the flag denoting the type of public key that we are using. Currently, we are only support edDSA, which has the flag set as 0x00, but in the future, ecDSA may have a flag set at 0x01.

**Comment by punwai** [comment]
I also checked out the json-rpc, but that doesn't seem affected as the docs are built from the RPC code itself, which has been updated. I think that one thing to say `Flag of the signature scheme that is used. (0x00 for edDSA [default])` rather than just `Flag of the signature scheme that is used`

**Comment by punwai** [comment]
TLDR: unless we explain address derivation - I don't think we have to worry about docs for now.

**Comment by Clay-Mysten** [comment]
> TLDR: unless we explain address derivation - I don't think we have to worry about updating the docs on this PR for now.

Thanks, @punwai ! I appreciate your help immensely. The docs are growing large and the changes many. So the more direct guidance the better.

If you think it's useful to anyone, we can note the change and link to the associated PR from:
https://docs.sui.io/learn#see-whats-new

For example, the recent:
sui::id is now sui::object and VersionedID is now Info. Use the [Object module](https://github.com/MystenLabs/sui/blob/main/crates/sui-framework/sources/object.move) that has [replaced the former ID.move.](https://github.com/MystenLabs/sui/pull/3241)

Otherwise, it seems we can close this. I leave that up to you. Thanks again!
